Support enum in attribute arguments

apple-x-co opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced new annotations `FakeMarker5` and `FakeMarker6` for method annotations.
  - Added PHP 8.1 enum `FakePhp81Enum` with cases `Apple`, `Orange`, and `Grape`.

- **Enhancements**
  - Updated `testVariousMethodSignature` to include additional assertions for new methods.
  - Refactored method signature handling for improved code readability and maintainability.

- **Bug Fixes**
  - Fixed loop condition in tests to cover additional method signatures.

- **Documentation**
  - Updated method annotations to reflect new enum cases and annotations.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

apple-x-co committed:
test: Add red test

apple-x-co committed:
fix: Method signature string

apple-x-co committed:
test:

apple-x-co committed:
test:

apple-x-co committed:
refactor: Method signature string

apple-x-co committed:
test:

apple-x-co committed:
test:

apple-x-co committed:
refactor: Method signature string

apple-x-co committed:
refactor: Method signature string

apple-x-co commented:
@koriym ご確認をよろしくお願いいたします。

koriym committed:
Refactor representArg method in MethodSignatureString

Simplified the representArg method logic for handling UnitEnum instances. The refactoring involved removing duplicate code to improve readability and maintainability.

koriym committed:
Refactor magic values in MethodSignatureString

Replaced magic values in MethodSignatureString class with class constants for better readability and maintainability. This includes PHP version check and the indent string.

koriym committed:
Refactor method signature string generation

The method signature string generation was refactored for improved readability and maintainability. Separate helper methods were created to handle the generation of doc comments, attributes, access modifiers, and the method signature itself. This reduces the complexity and length of the main 'get' method.

koriym committed:
Remove SuppressWarnings annotation from MethodSignatureString class

The SuppressWarnings PHPMD.CyclomaticComplexity annotation used on the MethodSignatureString class was not necessary and has now been removed. The removal of this annotation will ensure that the cyclomatic complexity app metric is appropriately analyzed and followed for this class.

koriym committed:
Change visibility of generateParameterCode method to private

The generateParameterCode method in MethodSignatureString.php has been changed from public to private for encapsulation purposes. This change increases the integrity of the data and ensures that

koriym committed:
Update representArg method return type

The return type of the representArg function in MethodSignatureString.php was updated. The previous nullable string type has been replaced with a non-nullable string type to ensure it always returns a string.

koriym committed:
Rename method from "representArg" to "formatArg"

The method in the MethodSignatureString class has been renamed from "representArg" to "formatArg" for better understanding of its function. This change improves readability and provides clearer naming convention in the codebase.

koriym committed:
Soothe phpstan

Update MethodSignatureString to enhance argument formatting

This commit updates the formatArg function in MethodSignatureString to ensure strict string output. It also removes unnecessary string casting in the foreach loop that processes argument lists. The changes provide a more consistent and reliable argument format for use in function signatures.

koriym commented:
@apple-x-co 少しリファクタリングするつもりが大きなものになってしまいました><
その代わり、`PHPMD.CyclomaticComplexity`警告抑制を取り除くことができました。

koriym commented:
@coderabbitai review

koriym commented on src/MethodSignatureString.php:
```diff
@@ -53,6 +54,14 @@ public function get(ReflectionMethod $method): string
+                        $formattedArgs[] = $argRepresentation;
+
+                        continue;
+                    }
```
@coderabbitai Is this still valid?

koriym committed:
Update array formatting and whitespace handling in code

Removed unnecessary spaces in array definitions in tests and updated the method of handling whitespace in the MethodSignatureString. This change enhances code readability by ensuring consistent formatting standards across the codebase.

koriym committed:
Refactor method attribute fetching process

The method attribute fetching process has been refactored to improve readability and maintainability. The method attributes are now fetched once and stored in a variable before being iterated over, rather than fetching them directly during the iteration. The 'psalm-suppress MixedAssignment' annotation has been replaced by a 'var mixed' comment, providing easier understanding of the variable's expected value format.

koriym commented on src/MethodSignatureString.php:
```diff
@@ -19,87 +20,108 @@
 
-        // アクセス修飾子を取得
+        $signatureParts[] = self::INDENT;
+    }
```
@coderabbitai [1d15b6d](https://github.com/ray-di/Ray.Aop/pull/212/commits/1d15b6dc741342af80ca8b9233af9b9a3d65f1df) enough?

koriym replied:
`@psalm-pure` is removed.

koriym replied:
@coderabbitai Yes, the problem was that I annotated it when I was not originally qualified to declare it `@psalm-pure`.

koriym commented on src/MethodSignatureString.php:
```diff
@@ -19,87 +20,108 @@
+     * @param mixed      $value
+     */
+    private function formatArg($name, $value): string
+    {
+        $formattedValue = $value instanceof UnitEnum ?
+            '\\' . var_export($value, true)
+            : preg_replace('/\s+/', '', var_export($value, true));
 
-        return implode(' ', $signatureParts);
+        return is_numeric($name) ? (string) $formattedValue : "{$name}: {$formattedValue}";
```
The original regular expression is sufficient.
